### PR TITLE
[FLINK-32977] Deprecate IOReadableWritable serialization in Path

### PR DESCRIPTION
**Pull request by WencongLiu**
## What is the purpose of the change

Deprecate IOReadableWritable serialization in Path.

## Brief change log

  - Deprecate IOReadableWritable serialization in Path

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Review by lsyldliu — APPROVED** [review]
@WencongLiu Thanks for contribution, LGTM

File: flink-core/src/main/java/org/apache/flink/core/fs/Path.java [hunk 1]
```diff
@@ -526,4 +528,55 @@ private boolean hasWindowsDrive(String path, boolean slashed) {
...
     }
+
+    /**
+     * Deserialize the Path from {@link DataInputView}.
+     *
+     * @param in the data input view.
+     * @return the path
+     * @throws IOException if an error happened.
+     */
+    public static Path deserializeFromDataInputView(DataInputView in) throws IOException {
```
Comment (xintongsong) [inline flink-core/src/main/java/org/apache/flink/core/fs/Path.java:539]: Should deduplicate, by calling `deserializeFromDataInputView` from `read`.

File: flink-core/src/main/java/org/apache/flink/core/fs/Path.java [hunk 2]
```diff
@@ -526,4 +528,55 @@ private boolean hasWindowsDrive(String path, boolean slashed) {
...
+            final String query = StringUtils.readNullableString(in);
+            final String fragment = StringUtils.readNullableString(in);
+
+            try {
+                result = new Path(new URI(scheme, userInfo, host, port, path, query, fragment));
+            } catch (URISyntaxException e) {
+                throw new IOException("Error reconstructing URI", e);
+            }
+        }
+        return result;
```
Comment (xintongsong) [inline flink-core/src/main/java/org/apache/flink/core/fs/Path.java:557]: Shall we return `null` if `isNotNull` is false?

Comment (xintongsong) [reply flink-core/src/main/java/org/apache/flink/core/fs/Path.java:557]: Actually, I wonder if we should forbid null values.

**Review by xintongsong — APPROVED** [review]
LGTM. Merging.

**Pull request closed without merging**